bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/PersistentInbox.java

```diff
@@ -171,7 +166,7 @@ public Thing approve(ThingUID thingUID, String label) {
     public synchronized boolean add(DiscoveryResult result) throws IllegalStateException {
         if (result != null) {
             ThingUID thingUID = result.getThingUID();
-            Thing thing = this.thingRegistry.get(thingUID);
+            Thing thing = getExistingThingForDiscoveryResult(result);
```

Comment:
Hm. This change looks against the discussed approach in a few ways:
- It changing the existing behavior by not at all adding results, although no Thing with the same UID exists. I thought that the idea was to have the "auto-ignore" as a configurable feature that does not interfere with the existing logic.
- If it is considered to be "auto-ignore", then it is wrong here, because it does not at all add the result. This make the whole feature unsymmetric: A result is present, but ignored, if the Discovery Result comes first and then the Thing is added. It is NOT present, if the Thing is added manually and then it is discovered.

Wouldn't it be much cleaner if the AutomaticInboxProcessor would also listen to Inbox events and just do the ignore functionality in either case?